R/columnmulticolor.R

```diff
@@ -134,10 +135,7 @@ ColumnMultiColor <- function(x,
         marker.border.opacity <- opacity
     else if (is.null(marker.border.opacity)) # trying to hide gap in the middle
         marker.border.opacity <- opacity/(4 + 3*(opacity < 0.7))
-    colors <- paste0(rep("", NROW(chart.matrix)), colors)
-    ind.na <- which(is.na(chart.matrix))
-    if (length(ind.na) > 0)
-        colors <- colors[-ind.na]
+    eval(colors)
```

Comment:
This line seems to be the opposite of changes in other plot types. i.e. `eval(colors)` is replaced with `colors <- paste0(rep("", NROW(chart.matrix)), colors)` except for here where the opposite occurs. Is there a reason for this? What is this `colors <- paste0(rep("", NROW(chart.matrix)), colors)` doing?

Reply:
Thanks for picking that up, I've changed it now to be the same as the others now. `colors <- paste0(rep("", NROW(chart.matrix)), colors)` is forcing the color vector to recycle (or trim down) to match the number of bars in the chart. We don't need to remove `ind.na` from colors because we've changed the way we handle missing values to leave a gap to show they are there.